src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/EntrypointResolver.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Solution
+{
+    public class EntrypointResolver : IEntrypointResolver
+    {
+        public IEnumerable<IProject> GetEntrypoints(IEnumerable<IProject> projects, IReadOnlyCollection<string> names)
+        {
+            if (projects is null)
+            {
+                throw new ArgumentNullException(nameof(projects));
+            }
+
+            if (names is null)
+            {
+                throw new ArgumentNullException(nameof(names));
+            }
+
+            var matcher = new Matcher(StringComparison.OrdinalIgnoreCase);
+
+            matcher.AddIncludePatterns(names);
+
+            var count = 0;
+            var firstProject = default(IProject);
+
+            foreach (var project in projects)
```

Comment:
It took me a few minutes to figure out what's going on here. This method could benefit from some comments.